File: backnetby/ProductsService/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ProductsService.Models
{
     public class Product
     {
          public int Id { get; set; }

          [Required, StringLength(150)]
          public string Name { get; set; } = default!;

          [StringLength(250)]
          public string? Description { get; set; }

          [Required, StringLength(100)]
          public string Category { get; set; } = default!;

          [StringLength(500)]
          public string? Image { get; set; }

          [Range(0, double.MaxValue)]
          public decimal Price { get; set; }

          [Range(0, int.MaxValue)]
          public int Stock { get; set; }
     }
}

File: frontnetby/src/services/types/product.ts
export type Product = {
     id: number;
     name: string;
     description: string | null;
     category: string;
     image: string | null;
     price: number;
     stock: number;
};

export type ProductCreate = {
     name: string;
     description: string | null;
     category: string;
     image: string | null;
     price: number;
     stock: number;
};

export type ProductUpdate = {
     name: string;
     description: string | null;
     category: string;
     image: string | null;
     price: number;
     stock: number;
};